File: src/main/java/kitchenpos/menu/tobe/application/dto/request/CreateMenuRequest.java
package kitchenpos.menu.tobe.application.dto.request;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record CreateMenuRequest(BigDecimal price,
                                String name,
                                UUID menuGroupId,
                                boolean display,
                                List<CreateMenuProductRequest> menuProducts) {
}

File: src/main/kotlin/kitchenpos/tobe/menu/application/dto/request/CreateMenuRequest.kt
package kitchenpos.tobe.menu.application.dto.request

import java.math.BigDecimal
import java.util.*

data class CreateMenuRequest(
    val name: String,
    val price: BigDecimal,
    val menuGroupId: UUID,
    val displayed: Boolean,
    val menuProducts: List<MenuProduct>,
) {
    data class MenuProduct(
        val productId: UUID,
        val quantity: Long,
    )
}
